feat: migrate`recreateClosed` to `recreateWhen` (#21039)
renovatebot/renovate: docs/usage/key-concepts/pull-requests.md

### How to fix immortal PRs

Avoid grouping dependencies together which have different versions, or which you have a high chance of wanting to ignore.

> If you still get immortal PRs then you can set `"recreateClosed": "never"`

Reviewer: ```suggestion
If you have immortal PRs which you'd like to remain closed you can set `"recreateClosed": "never"`.
```

> If you have immortal PRs which you'd like to remain closed you can set `"recreateClosed": "never"`.

Reviewer 2: ```suggestion
If you have immortal PRs which you want to keep closed, then set `"recreateClosed": "never"`.
```

Maybe something like this is easier to understand?

### How to fix immortal PRs

Avoid grouping dependencies together which have different versions, or which you have a high chance of wanting to ignore.

> If you have immortal PRs which you want to keep closed, then set `"recreateWhen": "never"`.

Reviewer: It depends whether you want it applied to every PR or not. There's certainly a case where people may decide they never want immortal PRs recreated

> If you have immortal PRs which you want to keep closed, then set `"recreateWhen": "never"`.

Reviewer 2: Maybe we need text for these cases then?

- Ignore a immortal update for a certain package(s)
- Ignore all immortal PRs

> If you have immortal PRs which you want to keep closed, then set `"recreateWhen": "never"`.

Reviewer: I don't think it's necessary, because "you can apply settings across the repo or via package rules" is a common concept and we don't want to repeat it for every config option we have.
